src/cell.cpp

```diff
@@ -235,5 +236,49 @@ bool cell_basic_equality(cell const& lhs, cell const& rhs)
     return true;
 }
 
+// Construct cell from flat morphology specification.
+
+cell make_cell(const morphology& morph) {
+    using point3d = cell::point_type;
+    cell newcell;
+
+    if (!morph) {
+        return newcell;
+    }
+
+    EXPECTS(morph.check_invariants());
+
+    // (not supporting soma-less cells yet)
+    newcell.add_soma(morph.soma.r, point3d(morph.soma.x, morph.soma.y, morph.soma.z));
+
+    for (const section_geometry& section: morph.sections) {
+        segmentKind kind = segmentKind::none;
```

Comment:
could this be a bit easier to understand as follows?
```
auto kind = section.kind;
if (kind==segmentKind::none) {
  kind = segmentKind::dendrite;
}
if (kind==segmentKind::soma) {
  throw std::invalid_argument("no support for complex somata");
}
```
This makes it clearer that kind is being assigned `section.kind`, then being filtered.